com.unity.cinemachine/Editor/Upgrader/CinemachineUpgradeManager.cs

```diff
@@ -113,204 +113,47 @@ public static void UpgradeProject()
             {
                 var manager = new CinemachineUpgradeManager();
 
-                manager.MakeTimelineNamesUnique(out var renames);  
-                manager.UpgradeInScenes(); // non-prefabs first
-                manager.UpgradePrefabsAndPrefabInstances();
-                manager.UpgradeAnimationTrackReferences();
-                manager.DeleteObsoleteComponentsInScenes();
+                var renames = manager.MakeTimelineNamesUnique();
+                manager.UpgradeNonPrefabReferencables();
+                var conversionLinksPerScene = manager.CopyPrefabInstances();
+                manager.UpgradePrefabAssets();
+                manager.UpgradeAllScenes(conversionLinksPerScene);
                 manager.RestoreTimelineNames(renames);
             }
         }
 
-        /// <summary>
-        /// We need to be able to tell timelines apart, and their name may not be unique. So we make them
-        /// unique here, and restore their original name when we are done with RestoreTimelineNames().
-        /// </summary>
-        /// <param name="renames">Dictionary, Key = GUID name generated, Value = original name</param>
-        void MakeTimelineNamesUnique(out Dictionary<string, string> renames)
-        {
-            renames = new Dictionary<string, string>();
-#if CINEMACHINE_TIMELINE
-            for (var s = 0; s < m_SceneManager.SceneCount; ++s)
-            {
-                var scene = OpenScene(s);
-                var directors = TimelineManager.GetPlayableDirectors(scene);
-                foreach (var director in directors)
-                {
-                    var originalName = director.name;
-                    director.name = GUID.Generate().ToString();
-                    renames.Add(director.name, originalName); // key = guid, value = originalName
-                }
-                EditorSceneManager.SaveScene(scene);
-            }
-#endif
-        }
-        
-        void RestoreTimelineNames(Dictionary<string, string> renames)
+        void UpgradeNonPrefabReferencables()
         {
-#if CINEMACHINE_TIMELINE
             for (var s = 0; s < m_SceneManager.SceneCount; ++s)
             {
                 var scene = OpenScene(s);
-                var directors = TimelineManager.GetPlayableDirectors(scene);
-                foreach (var director in directors)
+                var upgradables = 
+                    GetUpgradables(scene.GetRootGameObjects(), UpgradeObjectToCm3.Referencables, false);
+                foreach (var upgradable in upgradables)
                 {
-                    if (renames.ContainsKey(director.name)) // search based on guid name
+                    if (PrefabUtility.IsPartOfAnyPrefab(upgradable)) 
+                        continue;
+                    
+                    foreach (var referencable in UpgradeObjectToCm3.Referencables)
```

Comment:
This is weird.  Why are you looping again?  If upgradable contains multiple upgradable components, UpgradeObjectComponents() will handle it.  No need to upgrade multiple times.

Reply:
Yes. Sorry! Good catch! 👍  
The reason why I refactored GetUpgradables was to remove this part. Originally, I had a quick hack here just to filter out anything that's not referable.